### PR TITLE
Fix h2 connection leak

### LINKED ISSUE #19: Connection leak detection triggered for conn18
```
[15:06:40] [HikariPool-2 housekeeper/WARN] [com.zaxxer.hikari.pool.ProxyLeakTask]: Connection leak detection triggered for conn18: url=jdbc:h2:/home/container/./config/rankup/h2/playerstats user=, stack trace follows
java.lang.Exception: Apparent connection leak detected
at com.minecolonies.rankup.util.AccountingUtils.getConn(AccountingUtils.java:96) ~[AccountingUtils.class:?]
at com.minecolonies.rankup.util.AccountingUtils.createTableIfNeeded(AccountingUtils.java:147) ~[AccountingUtils.class:?]
at com.minecolonies.rankup.modules.databases.DatabaseModule.Rankup2Enable(DatabaseModule.java:22) ~[DatabaseModule.class:?]
at com.minecolonies.rankup.qsml.modulespec.StandardModule.onEnable(StandardModule.java:92) ~[StandardModule.class:?]
at uk.co.drnaylor.quickstart.loaders.ModuleEnabler.enableModule(ModuleEnabler.java:51) ~[ModuleEnabler.class:?]
at uk.co.drnaylor.quickstart.ModuleContainer$EnablePhase$2.onModuleAction(ModuleContainer.java:856) ~[ModuleContainer$EnablePhase$2.class:?]
at uk.co.drnaylor.quickstart.ModuleContainer.loadModules(ModuleContainer.java:476) ~[ModuleContainer.class:?]
at com.minecolonies.rankup.Rankup.onInit(Rankup.java:135) ~[Rankup.class:?]
at org.spongepowered.common.event.listener.GameInitializationEventListener_Rankup_onInit5.handle(Unknown Source) ~[?:?]
at org.spongepowered.common.event.RegisteredListener.handle(RegisteredListener.java:95) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:383) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at org.spongepowered.mod.event.SpongeModEventManager.extendedPost(SpongeModEventManager.java:443) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:404) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:231) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at org.spongepowered.mod.SpongeMod.onStateEvent(SpongeMod.java:271) ~[spongeforge-1.12.2-2705-7.1.0-BETA-3358.jar:1.12.2-2705-7.1.0-BETA-3358]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_171]
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_171]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.EventBus.post(EventBus.java:217) ~[minecraft_server.1.12.2.jar:?]
at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:218) ~[forge-1.12.2-14.23.4.2705-universal.jar:?]
at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:196) ~[forge-1.12.2-14.23.4.2705-universal.jar:?]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_171]
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_171]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116) ~[minecraft_server.1.12.2.jar:?]
at com.google.common.eventbus.EventBus.post(EventBus.java:217) ~[minecraft_server.1.12.2.jar:?]
at net.minecraftforge.fml.common.LoadController.redirect$onPost$zza000(LoadController.java:560) ~[LoadController.class:?]
at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:135) ~[LoadController.class:?]
at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:744) ~[Loader.class:?]
at net.minecraftforge.fml.server.FMLServerHandler.finishServerLoading(FMLServerHandler.java:107) ~[FMLServerHandler.class:?]
at net.minecraftforge.fml.common.FMLCommonHandler.onServerStarted(FMLCommonHandler.java:337) ~[FMLCommonHandler.class:?]
at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:219) ~[nz.class:?]
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486) ~[MinecraftServer.class:?]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]

```

### PR DESCRIPTION
**Pull request by flamechair**
Tested and verified locally. With this change, the error message doesn't pop up in the console and all functionality that was testable works as expected. This fixes: https://github.com/ldtteam/Rankup/issues/19

### CODE REVIEW

File: src/main/java/com/minecolonies/rankup/util/AccountingUtils.java [hunk 1]
```diff
@@ -88,17 +88,25 @@ private String getURI()
...
+            if (conn == null || conn.isClosed())
             {
-                conn = getDataSource(uri).getConnection();
-            }
-            catch (SQLException e)
-            {
-                getPlugin().getLogger().warn("Could not get Connection", e);
+                String uri = getURI();
+                
+                try
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/rankup/util/AccountingUtils.java]: MAJOR Extract this nested try block into a separate method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1141)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 29 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [rankup#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup#): Add a 'package-info.java' file to document the 'rankup' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [api#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/api#): Add a 'package-info.java' file to document the 'api' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [internal#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/internal#): Add a 'package-info.java' file to document the 'internal' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [command#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/internal/command#): Add a 'package-info.java' file to document the 'command' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [configurate#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/internal/configurate#): Add a 'package-info.java' file to document the 'configurate' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [events#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/internal/events#): Add a 'package-info.java' file to document the 'events' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [listener#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/internal/listener#): Add a 'package-info.java' file to document the 'listener' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [core#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/modules/core#): Add a 'package-info.java' file to document the 'core' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [command#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/modules/core/command#): Add a 'package-info.java' file to document the 'command' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [config#](https://github.com/ldtteam/Rankup/blob/cc2568ba900211c078af7572c18d757b4bd04c1f/src/main/java/com/minecolonies/rankup/modules/core/config#): Add a 'package-info.java' file to document the 'config' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 29 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [rankup#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup#): Add a 'package-info.java' file to document the 'rankup' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [api#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/api#): Add a 'package-info.java' file to document the 'api' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [internal#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/internal#): Add a 'package-info.java' file to document the 'internal' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [command#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/internal/command#): Add a 'package-info.java' file to document the 'command' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [configurate#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/internal/configurate#): Add a 'package-info.java' file to document the 'configurate' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [events#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/internal/events#): Add a 'package-info.java' file to document the 'events' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [listener#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/internal/listener#): Add a 'package-info.java' file to document the 'listener' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [core#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/modules/core#): Add a 'package-info.java' file to document the 'core' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [command#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/modules/core/command#): Add a 'package-info.java' file to document the 'command' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [config#](https://github.com/ldtteam/Rankup/blob/1acc2c0add13e2a00770711269cd40b3af0f751e/src/main/java/com/minecolonies/rankup/modules/core/config#): Add a 'package-info.java' file to document the 'config' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by Asherslab** [comment]
Will need to fix the conflicts first thanks =D

**Pull request closed without merging**